### PR TITLE
Fix bug in extract_region for irregular grids

### LINKED ISSUE #94: extract_region preprocessor no longer works on irregular grids
Hi,

we recently changed the irregular grid section of the `extract_region` preprocessor in `preprocessor/_area.py` from this:

```
    # irregular grids
    lats = cube.coord('latitude').points
    lons = cube.coord('longitude').points
    mask = np.ma.array(cube.data).mask
    mask += np.ma.masked_where(lats < start_latitude, lats).mask
    mask += np.ma.masked_where(lats > end_latitude, lats).mask
    mask += np.ma.masked_where(lons > start_longitude, lons).mask
    mask += np.ma.masked_where(lons > end_longitude, lons).mask
    cube.data = np.ma.masked_where(mask, cube.data)
    return cube
```

to this:
```
    # irregular grids
    lats = cube.coord('latitude').points
    lons = cube.coord('longitude').points
    select_lats = start_latitude < lats < end_latitude
    select_lons = start_longitude < lons < end_longitude
    selection = select_lats & select_lons
    data = da.ma.masked_where(~selection, cube.core_data())
    return cube.copy(data)
```

The new code doesn't work for irregular grids. It produces the error message:
```
  File "/users/modellers/ledm/workspace/ESMValToolTest/ESMValTool/esmvalcore/preprocessor/_area.py", line 82, in extract_region
    select_lats = start_latitude < lats < end_latitude
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

```
Any suggestions on how to fix it and keep it lazy? For now, I've reverted my local code back to the previous version. (but I'm not going to push it!)

The reason that we didn't catch the problem at the time is because the ocean recipes that use this were designed to demonstrate the diagnostics, not test the preprocessor. I'm changing recipe_ocean_example.yml to include an irregular grid CMIP5 model for future testing.

### PR DESCRIPTION
**Pull request by bouweandela**
Closes #94

### CODE REVIEW

**Comment by ledm** [comment]
> Hi Lee,
> 
> Good point about the negative longitudes, I'll see what I can do. Looking at the code you posted here [#94 (comment)](https://github.com/ESMValGroup/ESMValCore/issues/94#issue-456883767), I doubt that it worked before.

I can assure you that this was working before in the numpy version, but I suspect that it did not work in the first dask version.

File: tests/unit/preprocessor/_area/test_area.py [hunk 1]
```diff
@@ -146,5 +147,78 @@ def test_extract_named_region(self):
...
+        data,
+        var_name='tos',
+        units='K',
+        dim_coords_and_dims=dim_coord_spec,
+        aux_coords_and_dims=aux_coord_spec,
+    )
+    return cube
+
+
+def test_extract_region_irregular(irregular_grid_cube):
```
Comment (ledm) [inline tests/unit/preprocessor/_area/test_area.py]: We should also have a test which uses extracts negative longitudes.

Comment (bouweandela) [reply tests/unit/preprocessor/_area/test_area.py]: Implemented

**Comment by bouweandela** [comment]
@ledm I ran you recipe with the latest changes and the plot now looks like this: 
diag_CMIP5_MPI-ESM-LR_Omon_historical_r1i1p1_tos_prep_map_4_diag_map_4_2001_2003_map_0

**Comment by ledm** [comment]
Great work, I can confirm that I'm seeing the same result here too! Nice work @bouweandela!
